### PR TITLE
fix(windows): launch AppContainer processes reliably

### PR DESCRIPTION
**Pull request by Hydr0gen1**
## Summary

Fix Windows AppContainer launches failing with `CreateProcessW` error 203 and preserve the requested working directory for console programs.

## Root cause

- AppContainer process creation requires `LOCALAPPDATA` in an explicit environment block. The Windows launcher replaced the parent environment but omitted it, causing `ERROR_ENVVAR_NOT_FOUND` (203) whenever mounts selected the AppContainer path.
- `Path::canonicalize()` returns `\\?\` verbatim paths on Windows. Passing one as `lpCurrentDirectory` makes `cmd.exe` interpret the drive path as UNC and fall back to the Windows directory.

## Fix

- Set `LOCALAPPDATA` to the sandbox temp directory (no host profile disclosure).
- Convert verbatim disk/UNC prefixes to regular Win32 form only for `lpCurrentDirectory`.
- Add focused unit coverage.

## Validation

On Windows 11, before:

```text
arapuca run: launch failed: process: CreateProcessW: The system could not find the environment option that was entered. (os error 203)
```

After, a read-only AppContainer mount successfully runs `cmd.exe`, and `%CD%` is the requested directory. `cargo test platform::windows::tests` passes (16 tests).

### CODE REVIEW

**Review by sergio-correia — COMMENTED** [review]
One issue to address before merging, plus a minor edge case.

File: src/platform/windows.rs [hunk 1]
```diff
@@ -886,6 +881,10 @@ fn build_env(cfg: &Config, tmp_dir: &Path) -> Vec<(String, String)> {
     let system_root = std::env::var("SystemRoot").unwrap_or_else(|_| r"C:\Windows".into());
 
     env.push(("USERPROFILE".into(), tmp_dir.to_string_lossy().into_owned()));
+    env.push((
```
Comment (sergio-correia) [inline src/platform/windows.rs]: `filter_caller_env` doesn't block `LOCALAPPDATA`, `PATH`, `SystemRoot`, `TEMP`, `TMP`, or `USERPROFILE`. When a caller forwards these from the parent's environment (common for SDKs), they appear before the sandbox's hardcoded values after the stable sort in `encode_env_block`. On Windows, `GetEnvironmentVariable` returns the first match, so the caller's value silently wins.

Concretely: an FFI consumer forwards the host's `LOCALAPPDATA=C:\Users\...\AppData\Local`. It passes the filter, sorts before the sandbox's `LOCALAPPDATA=<tmp_dir>`, and the AppContainer has no DACL for that path. CreateProcessW fails with error 203 again.

Also, `Config::env` doc says "the last value wins" but the implementation does the opposite (first wins via stable sort + linear scan).

Please strip managed keys from the caller env before appending:
```rust
const OVERRIDDEN: &[&str] = &[
    "USERPROFILE", "LOCALAPPDATA", "TEMP", "TMP",
    "PATH", "SystemRoot", "LANG",
];
env.retain(|(k, _)| {
    !OVERRIDDEN.iter().any(|o| k.eq_ignore_ascii_case(o))
});
```

File: src/platform/windows.rs [hunk 2]
```diff
@@ -923,6 +922,28 @@ pub(crate) fn quote_args(cmd: &str, args: &[&str]) -> Vec<u16> {
...
+fn encode_work_dir(path: &Path) -> Vec<u16> {
+    use std::ffi::OsStr;
+
+    let mut wide: Vec<u16> = path.as_os_str().encode_wide().collect();
+    let verbatim = OsStr::new(r"\\?\").encode_wide().collect::<Vec<_>>();
+    let verbatim_unc = OsStr::new(r"\\?\UNC\").encode_wide().collect::<Vec<_>>();
+    if wide.starts_with(&verbatim_unc) {
+        wide.drain(..verbatim_unc.len());
+        wide.splice(0..0, OsStr::new(r"\\").encode_wide());
+    } else if wide.starts_with(&verbatim) {
```
Comment (sergio-correia) [inline src/platform/windows.rs]: This branch strips `\\?\` from any verbatim path, but `Path::canonicalize` can also return `\\?\Volume{GUID}\path` for volumes mounted without a drive letter. After stripping, the result is `Volume{GUID}\path`, which is not a valid Win32 path.

Only stripping when followed by a drive letter would be safer:
```rust
} else if wide.starts_with(&verbatim)
    && wide.len() > verbatim.len() + 2
    && (wide[verbatim.len()] >= b'A' as u16 && wide[verbatim.len()] <= b'z' as u16)
    && wide[verbatim.len() + 1] == b':' as u16
{
    wide.drain(..verbatim.len());
}
```

**Comment by Hydr0gen1** [comment]
Addressed both review threads in 34900ea:

- Removed caller-supplied `USERPROFILE`, `LOCALAPPDATA`, `TEMP`, `TMP`, `PATH`, `SystemRoot`, and `LANG` case-insensitively before appending launcher-managed values, so the sandbox values are the only entries encoded.
- Limited `\\?\` removal to drive-letter paths while preserving the existing UNC conversion; `\\?\Volume{GUID}\...` remains verbatim.
- Corrected the `Config::env` duplicate-key documentation for Windows behavior.
- Added regression tests for case-insensitive caller overrides and Volume GUID working directories.

Validation on Windows: 19 focused platform tests passed; full suite passed (161 tests); production-target clippy passed with warnings denied.

@sergio-correia, please re-review when convenient.

File: src/platform/windows.rs [hunk 3]
```diff
@@ -875,6 +870,20 @@ fn rollback_appcontainer(
 
 fn build_env(cfg: &Config, tmp_dir: &Path) -> Vec<(String, String)> {
     let mut env: Vec<(String, String)> = crate::env::filter_caller_env(&cfg.env).passed;
+    const OVERRIDDEN: &[&str] = &[
+        "USERPROFILE",
```
Comment (sergio-correia) [inline src/platform/windows.rs:874]: `APPDATA` (`%USERPROFILE%\AppData\Roaming`) should be in this list too. If a caller forwards it, the host username leaks into the sandbox environment, and programs that read `APPDATA` (Python's `site.py`, npm, etc.) get confusing errors pointing at inaccessible host paths.

Same treatment as `LOCALAPPDATA`: add to `OVERRIDDEN` and push the sandbox temp value below.

Lower priority but same class of leak: `HOMEPATH`, `HOMEDRIVE` (legacy `USERPROFILE` equivalent), and `USERNAME` (directly exposes the host account name).

**Review by sergio-correia — COMMENTED** [review]
The env override fix and verbatim path tightening both look correct. One more env var to add.

**Comment by Hydr0gen1** [comment]
Addressed the Jul 24 follow-up in a0a30ac:

- Added `APPDATA` to the overridden list and set it to the sandbox temp directory (same treatment as `LOCALAPPDATA`).
- Also overrode the same class of host-identity leaks: `HOMEDRIVE` / `HOMEPATH` (derived from the sandbox profile path) and `USERNAME` (fixed to `sandbox`).
- Added regression coverage for case-insensitive caller overrides of those keys.

Validation on Windows: focused env tests passed; full lib suite passed (163 tests); production-target clippy passed with warnings denied.

@sergio-correia, please re-review when convenient.

**Review by sergio-correia — APPROVED** [review]
Thanks, looks good to me!
